### PR TITLE
refactor: Extract quick jumping logic to ContainerQuickJumpModel

### PR DESCRIPTION
**Pull request by ShadowCat117**
Replaces https://github.com/Wynntils/Artemis/pull/2008

### CODE REVIEW

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.lwjgl.glfw.GLFW;
+
+public class ContainerQuickJumpModel extends Model {
+    private static final int NEXT_PAGE_SLOT = 8;
+    private static final int PREVIOUS_PAGE_SLOT = 17;
+    private static final List<Integer> BUTTON_SLOTS = List.of(7, 16, 25, 34, 43, 52);
+    private static final List<Integer> QUICK_JUMP_DESTINATIONS = List.of(1, 5, 9, 13, 17, 21);
+
+    public ContainerQuickJumpModel(ContainerModel containerModel) {
+        super(List.of(containerModel));
```
Comment (Mcrtin) [inline common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]: this model doesn't depend on ContainerModel

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    public void jumpToDestination(int currentPage, int pageDestination, int lastPage) {
```
Comment (Mcrtin) [inline common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]: this method needs a better name

**Review by Mcrtin — COMMENTED** [review]
This is basically #2008 without the automatic quick jump completion. I'm not saying it is worse but you'd have to write that system every time you want to jump somewhere so for consistency and easy use I'm leaning more towards #2008 on the other hand this is more adaptive (but I don't know what else you'd use this for.

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.containers.type;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]:
```suggestion
package com.wynntils.models.containers;
```

Move this to the suggested package.

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]: Depending on container model makes sense even if that doesn't happen at the moment. I am find with keeping or removing this.

**Comment by kristofbolyai** [comment]
> This is basically #2008 without the automatic quick jump completion. I'm not saying it is worse but you'd have to write that system every time you want to jump somewhere so for consistency and easy use I'm leaning more towards #2008 on the other hand this is more adaptive (but I don't know what else you'd use this for.

what? what is "automatic quick jump completion"?

**Comment by Mcrtin** [comment]
when you call jumpToDestination it doesn't actually jump there, it just jumps closer so you have to call it every time the Screen inits (that means that it may jumped) and you have to keep track of the location you want to jump to

**Comment by Mcrtin** [comment]
so you basically have to copy half of CustomBankPagesFeature just so you can quick-jump somewhere (there are 5 fields in the feature related to quick-jumping)

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java (hunk 3 above)
Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]: Oops

File: common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java (hunk 1 above)
Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/models/containers/type/ContainerQuickJumpModel.java]: Well now it fully depends on it :)

**Comment by ShadowCat117** [comment]
No longer needed since https://github.com/Wynntils/Artemis/pull/1994 is closed

**Pull request closed without merging**